### PR TITLE
Add some infantry and more polish

### PR DESCRIPTION
**Pull request by abcdefg30**
- Added the engineer
- Added allied/soviet attack dog
- Removed ```SpawnViceriod``` from ```^Infantry```
- Cleaned the voice file
- Cleaned the weapons file

### CODE REVIEW

File: rules/infantry.yaml [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+	#BombSight=4 ; detecting ivan's little friends
+	DetectCloaked:
+		CloakTypes: ivan
+		Range: 4
+
+DOG:
+	Inherits: ^Infantry
+	Buildable:
+		Queue: Infantry
+		BuildAtProductionType: Soldier
```
Comment (Phrohdoh) [inline rules/infantry.yaml]: I'd prefer we use `Infantry`, `Vehicle`, etc here instead.

File: weapons.yaml [hunk 2]
```diff
@@ -177,18 +177,18 @@ Sniper:
 			Heavy: 0
 			Concrete: 0
 
-Tiberium:
-	ReloadDelay: 16
+DogJaw:
+	ValidTargets: Infantry
+	ReloadDelay: 10
+	Range: 3c0
+	Report: DOGATTACK#DOGG5P.AUD
```
Comment (Phrohdoh) [inline weapons.yaml]: `DOGATTACK # DOGG5P.AUD` for this and others with comments, please.

File: rules/infantry.yaml (hunk 1 above)
Comment (abcdefg30) [inline rules/infantry.yaml]: Deleted that, as we don't need it (it was iirc for differing between kennel and barracks).

**Comment by abcdefg30** [comment]
Updated.

File: rules/defaults.yaml [hunk 3]
```diff
@@ -107,7 +107,7 @@
 	Selectable:
 		Voice: Infantry
 	TargetableUnit:
-		TargetTypes: Ground
+		TargetTypes: Ground, Infantry, Ivan
```
Comment (Phrohdoh) [inline rules/defaults.yaml]: `Ivan`??

File: rules/infantry.yaml [hunk 4]
```diff
@@ -0,0 +1,106 @@
+ENGINEER:
```
Comment (Phrohdoh) [inline rules/infantry.yaml]: Let's do these in all lowercase to be a bit future proof.

**Comment by abcdefg30** [comment]
Updated again.

File: rules/infantry.yaml [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+	Captures:
+		CaptureTypes: building, husk
+	TakeCover:
+	-AutoTarget:
+	RenderInfantry:
+		IdleAnimations: idle1,idle2
+		StandAnimations: stand
+	RevealsShroud:
+		Range: 4c0
+	DetectCloaked:
```
Comment (Phrohdoh) [inline rules/infantry.yaml]: What is the source for this?

File: rules/infantry.yaml [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+	RenderInfantry:
+		IdleAnimations: idle1,idle2
+	IgnoresDisguise:
+	DetectCloaked:
+		Range: 5
+# ImmuneToRadiation=no
+# ImmuneToPsionics=yes
+# IFVMode=0
+
+adog: # TODO: merge with ^
```
Comment (Phrohdoh) [inline rules/infantry.yaml]: Hmmm, consider `adog` and `sdog`? Or did RA2 have them as a single unit?

**Comment by abcdefg30** [comment]
Updated as discussed on IRC.

**Comment by Phrohdoh** [comment]
Will review when I return, thanks!

**Comment by Phrohdoh** [comment]
Note to self: Before reviewing this I need to update the sidebar code (removing the legacy `ChooseBuildTabOnSelect`).

**Comment by Phrohdoh** [comment]
Done! @abcdefg30 please rebase so I can merge this (and update #1 if you get the time please).

File: rules/infantry.yaml (hunk 6 above)
Comment (Phrohdoh) [inline rules/infantry.yaml]: We can merge this dogs into one now that OpenRA has [this](https://github.com/OpenRA/OpenRA/pull/7477).

**Comment by abcdefg30** [comment]
Updated.

**Comment by Phrohdoh** [comment]
Confirmed dog and engi for both factions.
Thanks for all of your work on this, merging!

**Pull request merged**